tests/test_client.py

```diff
@@ -1142,12 +1167,40 @@ async def disconnect(self) -> None:
         self.server.mass_inform("disconnect", "Server shutting down")
         await self.client.wait_disconnected()
 
+    @invariant()
+    def check_reading_invariant(self) -> None:
+        """Check that we don't have any readings stored for unsubscribed sensors."""
+        if self.client._sensor_monitor is not None:
+            for s in self.client._sensor_monitor._sensors.values():
+                assert s.subscribed or s.reading is None
+
+    @invariant()
+    def check_need_subscribe_invariant(self) -> None:
+        """Check that `_need_subscribe` is consistent with sensor state."""
+        if self.client._sensor_monitor is not None:
+            expected = set()
+            for s in self.client._sensor_monitor._sensors.values():
+                if (
+                    s.subscribed == _MonitoredSensor.Subscribed.UNKNOWN
+                    or (s.subscribed == _MonitoredSensor.Subscribed.YES and not s.watchers)
+                    or (s.subscribed == _MonitoredSensor.Subscribed.NO and s.watchers)
+                ):
+                    expected.add(s)
+            assert self.client._sensor_monitor._need_subscribe == expected
+
+    @invariant()
+    def check_state_invariant(self) -> None:
+        """Check that state is consistent with the update event."""
+        monitor = self.client._sensor_monitor
+        if monitor is not None:
+            assert (monitor._state == SyncState.SYNCING) == (monitor._update_event.is_set())
```

Comment:
This is quite the `assert` statement. Couple of questions
- Why the brackets around `monitor._update_event.is_set()`?
- Why this comparison/equating of two boolean results 
  - and not e.g. `(monitor_state == SyncState.SYNCING) and monitor._update_event.is_set()`?

Reply:
So an XNOR gate :-)